scripts/Tools/preview_namelists

```diff
@@ -15,50 +15,38 @@ from CIME.utils             import expect
 import argparse, doctest
 
 ###############################################################################
-def parse_command_line(args, description):
+def parse_command_line(args):
 ###############################################################################
-    parser = argparse.ArgumentParser(
-        usage="""\n%s [--debug]
-OR
-%s --verbose
-OR
-%s --help
-OR
-%s --test
-
-\033[1mEXAMPLES:\033[0m
-    \033[1;32m# Run \033[0m
-    > %s
-""" % ((os.path.basename(args[0]), ) * 5),
-
-description=description,
-
-formatter_class=argparse.ArgumentDefaultsHelpFormatter
-)
-
+    cime_model = CIME.utils.get_model()
+    parser = argparse.ArgumentParser()
     CIME.utils.setup_standard_logging_options(parser)
 
     parser.add_argument("caseroot", nargs="?", default=os.getcwd(),
                         help="Case directory to build")
+    parser.add_argument('--component',
+                        help="Specify component's namelist to build.")
+    parser.add_argument('--test', action='store_true',
+                        help="Run preview_namelist in test mode.")
 
-    args = parser.parse_args(args[1:])
+    args = parser.parse_args()
 
     CIME.utils.handle_standard_logging_options(args)
 
-    return args.caseroot
+    return args
 
 ###############################################################################
 def _main_func(description):
 ###############################################################################
-    if ("--test" in sys.argv):
+    args = parse_command_line(sys.argv)
```

Comment:
I agree that it's bad to tie user interface to lower-level code, but I feel this is worse:

```python
    output, build_dir, build_optimized, clean,\
        cmake_args, compiler, enable_genf90, machine, machines_dir,\
        make_j, use_mpi, mpilib, mpirun_command, test_spec_dir, ctest_args,\
        use_openmp, xml_test_list, verbose \
        = parse_command_line(sys.argv)
```

It's really hard to see if I have swapped some of the arguments in that list.

I agree that lower-level functions should accept specific arguments, but from the main program, it seems much better to me to replace the above with

```python
args = parse_command_line(sys.argv)
```

and then you can access the individual arguments as attributes of the args object. I do not see any downsides from either a practical standpoint or a philosophical elegance standpoint: If anything, I'd say that the latter is *less tied* to the user interface, because if there are arguments you don't care about, you don't have to list them explicitly here.